src/backend/src/services/onboarding.services.ts

```diff
@@ -321,6 +320,97 @@ export default class OnboardingServices {
     });
   }
 
+  /**
+   * Toggles a user's check on a checklist
+   * @param checklistId the id of the checklist to toggle
+   * @param userId the id of the user to toggle
+   * @returns the updated checklist
+   */
+  static async toggleChecklist(checklistId: string, userId: string) {
+    const checklist = await prisma.checklist.findUnique({
+      where: { checklistId },
+      include: { usersChecked: true, subtasks: { where: { dateDeleted: null }, include: { usersChecked: true } } }
+    });
+
+    if (!checklist) {
+      throw new NotFoundException('Checklist', checklistId);
+    }
+
+    if (checklist.dateDeleted) {
+      throw new DeletedException('Checklist', checklistId);
+    }
+
+    const isChecked = checklist.usersChecked.some((user) => user.userId === userId);
+
+    if (checklist.parentChecklistId == null && checklist.subtasks.length > 0) {
```

Comment:
why do we need the first part of this if statement, why not just if theres any unchecked subtasks should be sufficient